test/integration/targets/filter_core/tasks/main.yml

```diff
@@ -128,8 +135,15 @@
           - flat_one == [1, 2, 3, [4, [5]], 6, 7]
           - flat_two == [1, 2, 3, 4, [5], 6, 7]
           - flat_tuples == [1, 2, 3, 4]
+          - flat_full_null == [1, 'None', 3, 4, 5, 6, 7]
+          - flat_one_null == [1, 'None', 3, [4, [5]], 6, 7]
+          - flat_two_null == [1, 'None', 3, 4, [5], 6, 7]
```

Comment:
I'd add a new var (or change existing one) to have None below the first level to test the recursive call to flatten.